src/cli/cmds/synth.ts

```diff
@@ -13,31 +24,72 @@ class Command implements yargs.CommandModule {
     .option('stdout', { type: 'boolean', required: false, desc: 'Write synthesized manifests to STDOUT instead of the output directory', alias: 'p' })
-    .option('plugins-dir', { default: config.pluginsDirectory, required: false, desc: 'Directory to store cdk8s plugins.' })
-    .option('validate', { type: 'boolean', default: true, required: false, desc: 'Apply validation plugins on the resulting manifests (use --no-validate to disable)' })
-    .option('validation-reports-output-file', { required: false, desc: 'File to write a JSON representation of the validation reports to' });
-  ;
+    .option('plugins-dir', { required: false, desc: 'Directory to store cdk8s plugins.' })
+    .option('validate', { type: 'boolean', required: false, desc: 'Apply validation plugins on the resulting manifests (use --no-validate to disable)' })
+    .option('validation-reports-output-file', { required: false, desc: 'File to write a JSON representation of the validation reports to' })
+    .option('format', { required: false, desc: 'Synthesis format for Kubernetes manifests. The default synthesis format is plain kubernetes manifests.', choices: [SynthesisFormat.PLAIN, SynthesisFormat.HELM], type: 'string' })
+    .option('chart-api-version', { required: false, desc: 'Chart API version of helm chart. The default value would be \'v2\' api version when synthesis format is helm. There is no default set when synthesis format is plain.', choices: [HelmChartApiVersion.V1, HelmChartApiVersion.V2], type: 'string' })
+    .option('chart-version', { required: false, desc: 'Chart version of helm chart. This is required if synthesis format is helm.' });
 
   public async handler(argv: any) {
 
     const command = argv.app;
-    const outdir = argv.output;
     const stdout = argv.stdout;
-    const validate = argv.validate;
-    const pluginsDir = argv.pluginsDir;
+    const outdir = argv.output ?? config?.output ?? (!stdout ? DEFAULT_OUTPUT_DIR : undefined);
+    const validate = argv.validate ?? true;
     const reportFile = argv.validationReportsOutputFile;
+    const pluginsDir = argv.pluginsDir ?? config?.pluginsDirectory ?? DEFAULT_PLUGINS_DIR;
+    const format = argv.format ?? config?.synth?.format ?? SynthesisFormat.PLAIN;
+    const chartVersion = argv.chartVersion ?? config?.synth?.chartVersion;
+    const chartApiVersion = argv.chartApiVersion ?? config?.synth?.chartApiVersion ?? getDefaultChartApiVersion(format);
 
-    if (outdir && outdir !== config.output && stdout) {
+    if (outdir && outdir !== config?.output && stdout) {
       throw new Error('\'--output\' and \'--stdout\' are mutually exclusive. Please only use one.');
     }
 
     if (outdir) {
       fs.rmSync(outdir, { recursive: true, force: true });
     }
 
+    if (format != SynthesisFormat.PLAIN && format != SynthesisFormat.HELM) {
+      throw new Error(`You need to specify synthesis format either as ${SynthesisFormat.PLAIN} or ${SynthesisFormat.HELM} but received: ${format}`);
+    }
```

Comment:
I believe this is dead code because this validation already happens thanks to the `choices` option...double check? :)